File: program-manager_sheduler/api/src/main/java/in/stackroute/planage/promanager/model/Participant.java
package in.stackroute.planage.promanager.model;



import java.util.Date;

import java.util.List;
import java.util.UUID;

import in.stackroute.planage.promanager.model.ParticipantTask;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Document

public class Participant {
   @Id
  private String id =UUID.randomUUID().toString();
  private String name;
  private String email;
  private String programId;
  private String productId;
  private String text;
  private Date startDate;
  private Date endDate;
  private List<ParticipantTask> tasks;
  public Participant( String name, String email, String programId, String text, Date startDate, Date endDate,String productId) {
    this.name = name;
    this.email = email;
    this.programId = programId;
    this.text = text;
    this.startDate = startDate;
    this.endDate = endDate;
    this.productId=productId;
  }
}

File: program-manager_sheduler/promanager-ui/ui/src/app/model/participant.ts
export class CSVRecord {
  public name: string;
  public email: string;
}

export class Participant {
  public id: string;
  public name: string;
  public email: string;
  public programId: string;
  public productId: string;
  public text: string;
  public startDate: Date;
  public endDate: Date;
  public tasks: Array<Task>;
}

export class Task {
  public taskId: string;
  public taskName: string;
  public dueDate: Date;
  public status: boolean;
}

export class TaskCalendar {
  public taskId: string;
  public taskName: string;
  public endDate: Date;
  public startDate: Date;
  public status: boolean;
}

export class ParticipantList {
  public id: string;
  public name: string;
  public email: string;
  public programId: string;
  public productId: string;
  public text: string;
  public startDate: Date;
  public endDate: Date;
  public tasks: Array<Rseponse>;
}

export class Rseponse {
  public date: string;
  public tasks: Array<Task>;
}
